sklearn/model_selection/_plot.py

```diff
@@ -111,16 +245,23 @@ def plot(
             `scikit-learn`.
 
         score_name : str, default=None
-            The name of the score used to decorate the y-axis of the plot. If
-            `None`, the generic name "Score" will be used.
+            The name of the score used to decorate the y-axis of the plot.If
+            `None` and `scoring` is provided, the same will be inferred from it
+            by removing `"neg_"` and replacing underscore by spaces. If
+            `scoring is None`, we return the default value `"Negative score"`
+            if `negate_score` is `True` and `"Score"` otherwise.
 
         score_type : {"test", "train", "both"}, default="test"
             The type of score to plot. Can be one of `"test"`, `"train"`, or
             `"both"`.
```

Comment:
I think it's quite unfortunate to have chosen `default="test"` for learning curves. I think we almost always want to plot `score_type="both"` to conduct an underfitting/overfitting analysis on learning curves and validation curves.

I think we could change the default value for plotting utilities without having to go through a future warning cycle.

Reply:
I can make a follow-up PR dedicated for that.

Reply:
> I think we could change the default value for plotting utilities without having to go through a future warning cycle.

+1

Reply:
I made the change in this PR directly. I added an entry in 1.3 to acknowledge the "breaking" change and mention how to set the parameter to get the previous parameter.